python/ray/tune/analysis/experiment_analysis.py

```diff
@@ -539,6 +541,16 @@ def best_result(self) -> Dict:
                 "the metric and mode explicitly and fetch the last result.")
         return self.best_trial.last_result
 
+    def _delimiter(self):
+        delimiter = os.environ.get("TUNE_RESULT_DELIM", "/")
+        if delimiter == "/" and log_once("delimiter_deprecation"):
+            warnings.warn(
+                "Dataframes will use '/' instead of '.' to delimit "
+                "nested result keys in future versions of Ray. For forward "
+                "compatibility, set the environment variable "
+                "TUNE_RESULT_DELIM='.'")
```

Comment:
Maybe I'm confused here, but shouldn't this be the other way around?

I.e. we use

```
delimiter = os.environ.get("TUNE_RESULT_DELIM", ".")
```

because it is the current behavior. Then we check
```
if delimiter == "." and log_once("delimiter_deprecation"):
```

because the `.` will be deprecated, and then we suggest

```

                "nested result keys in future versions of Ray. For forward "
                "compatibility, set the environment variable "
                "TUNE_RESULT_DELIM='/'")
```

because `/` will be used in the future?